### PR TITLE
Initial refactor

### PR DESCRIPTION
**Pull request by carlosu7**
Remove public accessors to the following classes:
* Command
* Kind
* QueryType
* SupportCommand

Refactor ProjectModelFactory::withQueryType to withSafeQueryType and always set quertyType to QueryType.SAFE

### CODE REVIEW

**Comment by nahsra** [comment]
@carlosu7 can we fix the SonarCloud warning?

File: plugins/codemodder-plugin-maven/src/main/java/io/codemodder/plugins/maven/operator/POMScanner.java [hunk 1]
```diff
@@ -36,7 +49,7 @@ public class POMScanner {
    * @throws IOException If an I/O error occurs.
    * @throws URISyntaxException If there is an issue with the URI syntax.
    */
```
Comment (nahsra) [inline plugins/codemodder-plugin-maven/src/main/java/io/codemodder/plugins/maven/operator/POMScanner.java:51]: Should we add anther ticket/task along the way to convert `File` -> `Path`?

File: plugins/codemodder-plugin-maven/src/test/java/io/codemodder/plugins/maven/operator/POMOperatorVersionQueryTest.java [hunk 2]
```diff
@@ -20,13 +19,60 @@ final class POMOperatorVersionQueryTest {
   void testBasicQuery() {
     String pomFile = "pom-1.xml";
 
-    Arrays.stream(QueryType.values())
-        .filter(queryType -> queryType != QueryType.NONE)
+    Optional<VersionQueryResponse> optionalVersionQueryResponse = null;
```
Comment (nahsra) [inline plugins/codemodder-plugin-maven/src/test/java/io/codemodder/plugins/maven/operator/POMOperatorVersionQueryTest.java:22]: We can collapse these:

```
catch (ExceptionType1 | ExceptionType2 | ExceptionType3 e) {
 ...
}
```

File: plugins/codemodder-plugin-maven/src/main/java/io/codemodder/plugins/maven/operator/POMScanner.java (hunk 1 above)
Comment (carlosu7) [reply plugins/codemodder-plugin-maven/src/main/java/io/codemodder/plugins/maven/operator/POMScanner.java:51]: Ticket created

File: plugins/codemodder-plugin-maven/src/test/java/io/codemodder/plugins/maven/operator/POMOperatorVersionQueryTest.java [hunk 3]
```diff
@@ -20,13 +19,60 @@ final class POMOperatorVersionQueryTest {
...
+
+    VersionQueryResponse versionQueryResponse = optionalVersionQueryResponse.get();
+
+    Assert.assertTrue(
+        "Version defined is 1.8 as source", versionQueryResponse.getSource().satisfies("=1.8.0"));
+    Assert.assertTrue(
+        "Version defined is 1.8 as target", versionQueryResponse.getTarget().satisfies("=1.8.0"));
+  }
+
+  @Test
```
Comment (nahsra) [inline plugins/codemodder-plugin-maven/src/test/java/io/codemodder/plugins/maven/operator/POMOperatorVersionQueryTest.java:45]: On second thought, why are we catching any of these? Just let them propagate and fail the test.

File: plugins/codemodder-plugin-maven/src/test/java/io/codemodder/plugins/maven/operator/POMOperatorVersionQueryTest.java [hunk 4]
```diff
@@ -187,52 +122,71 @@ void testPomVersion3() {
...
+    String pomFile = "pom-version-7.xml";
 
-              Assert.assertTrue(
-                  "Version defined is 1.7 as source",
-                  versionQueryResponse.getSource().satisfies("=1.7.0"));
-              Assert.assertTrue(
-                  "Version defined is 1.8 as target",
-                  versionQueryResponse.getTarget().satisfies("=1.8.0"));
-            });
+    Optional<VersionQueryResponse> optionalVersionQueryResponse = null;
```
Comment (nahsra) [inline plugins/codemodder-plugin-maven/src/test/java/io/codemodder/plugins/maven/operator/POMOperatorVersionQueryTest.java:163]: Just propagate

**Review by nahsra — CHANGES_REQUESTED** [review]
Just propagate the exception failures

**Comment by carlosu7** [comment]
@nahsra exceptions propagated except the ones that are inside lambda functions, in those cases i collapsed them

**Review by nahsra — APPROVED** [review]

**Pull request merged**